lib/bindings/c/src/lib.rs

```diff
@@ -306,33 +315,593 @@ pub extern "C" fn dynamo_kv_event_publish_removed(
+            return DynamoLlmResult::ERR;
+        }
+    };
+
+    let result = wk.runtime().secondary().block_on(async {
+        let router = match KV_ROUTER.get() {
+            Some(router) => router,
+            None => {
+                eprintln!("KV Router not initialized - call dynamo_kv_router_init first");
+                return DynamoLlmResult::ERR;
+            }
+        };
+
+        // First, get or initialize the preprocessor
+        let preprocessor = match PREPROCESSOR.get() {
+            Some(preprocessor) => preprocessor,
+            None => {
+                eprintln!("Preprocessor not initialized - call dynamo_kv_router_init first");
+                return DynamoLlmResult::ERR;
+            }
+        };
+
+        // Tokenize the prompt
+        let encoding = match preprocessor.tokenize(prompt) {
+            Ok(encoding) => encoding,
+            Err(e) => {
+                eprintln!("Failed to tokenize prompt: {:?}", e);
+                return DynamoLlmResult::ERR;
+            }
+        };
+
+        let tokens = encoding.token_ids();
+        let num_tokens = tokens.len();
+
+        // This replicates the exact logic from the if query_instance_id block (no config override):
+        match router.find_best_match(context_id, tokens, None, false).await {
+            Ok((instance_id, _overlap_amount)) => {
+                // Return worker_instance_id
+                unsafe {
+                    *worker_instance_id_out = instance_id;
+                }
+
+                // Return the tokens (copy them to C-managed memory)
+                let tokens_copy = unsafe { libc::malloc(num_tokens * std::mem::size_of::<u32>()) } as *mut u32;
+                if tokens_copy.is_null() {
+                    eprintln!("Failed to allocate memory for tokens");
+                    return DynamoLlmResult::ERR;
+                }
+
+                unsafe {
+                    std::ptr::copy_nonoverlapping(tokens.as_ptr(), tokens_copy, num_tokens);
+                    *token_ids_out = tokens_copy;
+                    *token_count_out = num_tokens;
+                }
+
+                tracing::trace!(
+                    "Tokens requested in the response through the query_instance_id annotation: {:?}",
+                    tokens
+                );
+
+                // Auto-cleanup: Free the request since this is just a query/probe
+                router.free(context_id).await;
+
+                DynamoLlmResult::OK
+            }
+            Err(e) => {
+                eprintln!("Failed to find best match: {:?}", e);
+                DynamoLlmResult::ERR
+            }
+        }
+    });
+
+    result
+}
+
+/// Query the best worker instance for a given prompt with configuration overrides
+/// This function accepts per-request configuration overrides for fine-tuned routing
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn dynamo_kv_router_query_instance_id_with_config(
```

Comment:
This looks really similar to `dynamo_kv_router_query_instance_id`. Should one call the other?